tesseract_time_parameterization/src/time_optimal_trajectory_generation.cpp

```diff
@@ -277,13 +277,17 @@ class CircularPathSegment : public PathSegment
     }
 
     // directions must be different at this point so angle is always non-zero
+    // This does not match the paper which should be acos(start_direction.dot(end_direction))
     const double angle = acos(std::max(-1.0, start_direction.dot(end_direction)));
     const double start_distance = (start - intersection).norm();
     const double end_distance = (end - intersection).norm();
 
     // enforce max deviation
-    double distance = std::min(start_distance, end_distance);
-    distance = std::min(distance, max_deviation * sin(0.5 * angle) / (1.0 - cos(0.5 * angle)));
+    double l1 = start_distance * 0.5;
+    double l2 = end_distance * 0.5;
```

Comment:
The paper was multiplying by 0.5.

Reply:
Also this change makes the trajectory duration increase. Not sure why it is different. 

From the [paper](http://www.roboticsproceedings.org/rss08/p27.pdf):

>where  the  first  two  elements  give  the  maximum  possible distances such that the circular segment does not replace more than half of the adjoining linear segments and the last element limits  the  radius  to  make  sure  the  circlular  segment  stays within a distance δ from waypoint qi

Reply:
@mpowelson let me know if you have an issue with this change and I can change it back.

Reply:
It's possible that these changes were added by the move it folks. I'm not sure how closely this follows the original implementation.

Reply:
The PR which added it to MoveIt was pretty well structured. See https://github.com/ros-planning/moveit/pull/1365.

The first commit imports the upstream code, directly from the author of the paper. Later commits add the wrappers and MoveIt adapters.

Reply:
@mpowelson The original commit form the author had it without the 0.5 factor so I have reverted my change to keep it the same.

@gavanderhoorn Thank you for the information and I was able to found the commit which added the change `acos(std::max(-1,...` so I added the comments from the PR in the code for future readers if they compare back to the paper.

Reply:
I would perhaps even just straight link to the MoveIt PRs.

And if we fix something here, contribute it back.

Reply:
> I would perhaps even just straight link to the MoveIt PRs.

Done.

> And if we fix something here, contribute it back.

Ideally that would be preferred, though I personally don't have the bandwidth for this. Though my preference would be to create a separate repository for time parameterization and join resource with maintaining it since these are independent of Tesseract and MoveIt. Not sure if the maintainers of MoveIt would be open to this, but I would be.